fraud/proof.go

```diff
@@ -26,6 +26,8 @@ func (p ProofType) String() string {
 type Proof interface {
 	// Type returns the exact type of fraud proof
 	Type() ProofType
+	// Hash returns the block hash.
+	Hash() string
```

Comment:
I'm wondering if we should call this what it is: `BlockHash()`? Or maybe `BlockID()` even simply `ID()`? As it identifies the Proof to map it uniquely to a block and hashes are usually of type `[]byte`. 🤔

Reply:
+1 on just calling it `BlockHash()`.